docs/pages/access-controls/idps/saml-attribute-mapping.mdx

```diff
@@ -0,0 +1,194 @@
+the usage syntax and it's result, evaluated against the following reference user spec file:
+```
+# reference user spec file
+kind: user
+metadata:
+  name: foobar
+spec:
+  roles:
+    - access
+    - editor
+    - dev-ssh
+  traits:
+    aws_role_arns: null
+    firstname:
+      - foo
+    lastname:
+      - BAR
+    displayname:
+      - foo bar
+    email:
+      - foobar@example.com
+    groups:
+      - okta-admin
+      - dev-sso
+      - dev-rdp
+```
+### Methods
+#### `add`
+Add new value(s). Works on `user.spec.roles` and `user.spec.traits.groups`.
+Expression | Result
+-----------| ------
+`user.spec.roles.add("staging-ssh")` | `access, editor, dev-ssh, staging-ssh`
+
+You can also add an entirely new value that is not available in the user spec file.
+E.g.:
+Expression | Result
+-----------| ------
+`set.add("entirely-new-role")` | `entirely-new-role`
+
+#### `remove`
+Remove value(s). Works on `user.spec.roles` and `user.spec.traits.groups`
+Expression | Result
+---------- | ------
+`user.spec.roles.remove("editor", "access")` | `dev-ssh`
+
+#### `contains`
+Returns boolean value for matching expression. To be used in helper functions such as `ifelse`.
+Works on `user.spec.roles` and `user.spec.traits.groups`.
+Expression | Result
+---------- | ------
+`user.spec.traits.groups.contains("okta-admin")` | `true`
+
+### Helper functions
+#### `strings.upper`
+Transform string to upper.
+Expression | Result
+-----------| ------
+`strings.upper(user.spec.traits.firstname)` | `FOO`
+
+#### `strings.lower`
+Transform string to lower.
+Expression | Result
+-----------| ------
+`strings.lower(user.spec.traits.lastname)` | `bar`
+
+#### `strings.replaceall`
+Replace all matching strings.
+Expression | Result
+-----------| ------
+`strings.replaceall(user.spec.traits.groups, "-", "+")` | `okta+admin, dev+sso, dev+rdp`
+`strings.replaceall(user.spec.traits.groups, "admin", "dev")` | `okta-dev, dev-sso, dev-rdp`
+
+#### `strings.split`
+Split string at matching character.
+Expression | Result
+-----------| ------
+`strings.split(user.spec.traits.groups, "-", " ")` | `okta admin, dev sso, dev rdp`
+
+#### `ifelse`
+Conditionally return values. To be used in conjunction with methods such as `contains`.
+
+Signature: `ifelse(condition, "value to return if condition is true", "value to return if condition is false")`
+Expression | Result
+-----------| ------
+`ifelse(user.spec.traits.groups.contains("okta-admin"), user.spec.traits.groups.add("newgroup"), user.spec.traits.groups)` | `okta-admin, dev-sso, dev-rdp, newgroup`
+
+#### `union`
+Combine values in `user.spec.traits.groups` and `user.spec.roles`.
+Expression | Result
+-----------| ------
+`union(user.spec.traits.groups, user.spec.roles)` | `okta-admin, dev-sso, dev-rdp, access, editor, dev-ssh`
+`union(user.spec.traits.groups.remove("okta-admin"), user.spec.roles)` | `dev-sso, dev-rdp, access, editor, dev-ssh`
+
+
+## Testing attribute mapping
+### Prerequisites
```

Comment:
@ptgott I added Prerequisites here as the requirements are only for testing commands below. But lmk if we prefer to keep it at top of the page.

Reply:
I think prerequisites should go on top to be consistent with other guides.